dsn/dsn.go

```diff
@@ -203,18 +200,36 @@ func HidePassword(dsn string) string {
 	return dsn
 }
 
-func ParseSocketFromNetstat(out string) string {
-	lines := strings.Split(out, "\n")
-	for _, line := range lines {
-		if strings.HasPrefix(line, "unix") && strings.Contains(line, "mysql") {
-			fields := strings.Fields(line)
-			socket := fields[len(fields)-1]
-			if path.IsAbs(socket) {
-				return socket
+// GetSocketFromProcessLists will loop through the list of PIDs until it finds a process
+// named 'mysqld' and the it will try to get the socket by querying the open network
+// connections for that process.
+// Warning: this function returns the socket for the FIRST mysqld process it founds.
+// If there are more than one MySQL instance, only the first one will be detected.
+func GetSocketFromProcessLists() (string, error) {
+	pids, err := process.Pids()
+	if err != nil {
+		return "", errors.Wrap(err, "Cannot get the list of PIDs")
+	}
+	for _, pid := range pids {
+		proc, err := process.NewProcess(pid)
+		if err != nil {
+			continue
+		}
+		if procName, err := proc.Name(); err != nil {
+			continue
+		} else {
+			if procName == "mysqld" {
```

Comment:
@cfsalguero Are you sure `mysqlx.socket` will be excluded from the list? The same `mysqld` process opens two sockets, `/var/run/mysqld/mysqld.sock` and `/var/run/mysqld/mysqlx.sock`. The second one should be ignored. I would suggest to just filter out `mysqlx.sock` from the list if we can't detect this in any other way. Maybe we could try to connect to it and check for ErrInvalidConn error (https://github.com/go-sql-driver/mysql/blob/master/packets.go#L38), but I'm not sure, so filtering out might be fine enough. But of course, firstly would be nice to verify if we get more than one socket here?

We got such error when testing pmm-client with @Nailya . I wasn't able to reproduce it, but looks like issue with picking `mysqlx.sock` instead of `mysqld.sock`.
```
nailya.kutlubaeva@bm-debian9-pmmtest:~$ sudo pmm-admin add mysql 
[linux:metrics] OK, already monitoring this system.
[mysql] 2018/06/19 11:20:12 packets.go:36: unexpected EOF
[mysql] 2018/06/19 11:20:42 packets.go:36: unexpected EOF
[mysql] 2018/06/19 11:21:12 packets.go:36: unexpected EOF
[mysql:metrics] Cannot connect to MySQL: driver: bad connection
```
Same as for `sudo pmm-admin add mysql --socket=/var/run/mysqld/mysqlx.sock`, if we change to `sudo pmm-admin add mysql --socket=/var/run/mysqld/mysqld.sock` then it works.